pageserver/src/layered_repository.rs

```diff
@@ -1262,7 +1262,7 @@ impl LayeredTimeline {
         seg: SegmentTag,
         lsn: Lsn,
         self_layers: &MutexGuard<LayerMap>,
-    ) -> Result<Option<(Arc<dyn Layer>, Lsn)>> {
```

Comment:
Those were `anyhow::Result` already before, but I've decided to not to replace them all (removing the import), but only in places where I've replaced asserts.

IMO, `Result` should stay the `std::result::Result` from the std prelude, and the rest of the "external" results should be qualified for explicitness.